### PR TITLE
Gl: Use core context

### PR DESCRIPTION
**Pull request by vlj**
This PR makes GL uses a core context instead of a compatibility one.

We loose some feature like stippled line or polygon but I didn't see them outside of the primitive demo ; on the other hands it makes rpcs3 gl backend more likely to run on OSX. In addition it makes GL debug tool more reliable in my experience, Nvidia NSight is able to capture frame in GL mode now.

### CODE REVIEW

**Comment by kd-11** [comment]
Thanks for taking care of this. I was going to tackle it eventually, since even renderdoc refuses to debug compatibility profile.

**Comment by vlj** [comment]
Seems like travis-ci has an old wxwidget.

**Comment by tambry** [comment]
You'll have to modify the Travis script to download a newer wxWidgets package, if one's available for Ubuntu Precise.
We can't also use Travis's packaged wxWidgets, because they're very slow at approving stuff - https://github.com/travis-ci/apt-package-whitelist/issues/217 and https://github.com/travis-ci/apt-package-whitelist/issues/417.

**Comment by danilaml** [comment]
@vlj wxWidgets 3.1 has just released so it would be better to update both travis (and appveyor)  and our submodule (since it's in some middle commit between releases we had to use for vs2015 support).

File: rpcs3/Gui/GLGSFrame.cpp [hunk 1]
```diff
@@ -4,7 +4,21 @@
 
 GLGSFrame::GLGSFrame() : GSFrame("OpenGL")
 {
+	// For some reason WX_GL_MAJOR_VERSION, WX_GL_MINOR_VERSION and WX_GL_CORE_PROFILE are not defined on travis build despite using wxWidget 3.0+
```
Comment (paulsapps) [inline rpcs3/Gui/GLGSFrame.cpp]: Should be in glcanvas.h ...

**Comment by kd-11** [comment]
I've noticed some missing geometry in some games with this changeset. I'll try to track down the cause.

**Comment by vlj** [comment]
Will prepare some separate gl commit before resubmitting this.

**Comment by kd-11** [comment]
@vlj some of the issues are resolved by the reworked mipmap generation in https://github.com/RPCS3/rpcs3/pull/1540

**Pull request closed without merging**